docs/UsingCppInterOp.rst

```diff
@@ -67,33 +67,204 @@ that it was designed for.
+scientists that are working with legacy C++ code and would like to
+migrate to simpler, more interactive languages.
+
+The goal of this research is to eventually land these interoperability
+tools (including CppInterOp) to greater communities like LLVM and Clang,
+to enable C++ to interact with other languages besides Python.
+
+Example: Template Instantiation
+-------------------------------
+
+The developmental Cppyy version can run basic examples such as the one
+here. Features such as standalone functions and basic classes are also
+supported.
+
+C++ code (Tmpl.h)
+
+::
+
+   template <typename T>
+   struct Tmpl {
+     T m_num;
+     T add (T n) {
+       return m_num + n;
+   }
+   };
+
+Python Interpreter
+
+::
+
+   >>> import cppyy
+   >>> import cppyy.gbl as Cpp
+   >>> cppyy.include("Tmpl.h")
+   >>> tmpl = Tmpl[int]()
+   >>> tmpl.m_num = 4
+   >>> print(tmpl.add(5))
+   9
+   >>> tmpl = Tmpl[float]()
+   >>> tmpl.m_num = 3.0
+   >>> print(tmpl.add(4.0))
+   7.0
+
+Where does the cppyy code reside?
+---------------------------------
+
+Following are the main components where cppyy logic (with Compiler Research
+Organization’s customizations started by `sudo-panda`_) resides:
 
-**LLVM Integration:** The CppInterOp interfaces will be part of the LLVM 
-toolchain (as part of Clang-REPL).
+-  `cppyy <https://github.com/compiler-research/cppyy>`_
+-  `cppyy-backend <https://github.com/compiler-research/cppyy-backend>`_
+-  `CPyCppyy <https://github.com/compiler-research/CPyCppyy>`_
 
-**Better C++ Support:** C++ features such as Partial Template Specialisation will
-be available through CppInterOp.
+..
 
-**Fewer Lines of Code:** A lot of dependencies and workarounds will be removed,
-reducing the lines of code required to execute CPPYY.
+   Note: These are forks of the `upstream cppyy`_ repos created by `wlav`_.
 
-**Well tested interoperability Layer:** The CppInterOp interfaces have full
-unit test coverage.
+CppInterOp is a separate library that helps these packages communicate with C++
+code.
+
+-  `CppInterOp <https://github.com/compiler-research/CppInterOp/tree/main>`_
+
+How cppyy components interact with each other
+---------------------------------------------
+
+**cppyy-backend**
+
+The ``cppyy-backend`` forms a layer over ``cppyy``, modifying some
+functionality to provide the functions required for ``CPyCppyy``. It also adds
+some `utilities`_ to help with repackaging and redistribution.
+
+For example, it initializes the interpreter (using the
+``clingwrapper::ApplicationStarter`` function), adds the required
+``include`` paths, and adds the headers required for cppyy to work. It
+also adds some checks and combines two or more functions to help
+CPyCppyy work.
+
+These changes help ensure that any change in ``cppyy`` doesn’t directly
+affect ``CPyCppyy``, and the API for ``CPyCppyy`` remains unchanged.
+
+**CPyCppyy**
+
+``CPyCppyy`` uses the functionality provided by ``cppyy-backend`` and
```

Comment:
I guess I'm left asking why we need to introduce the details of cpycppyy at all..

Reply:
As discussed in cppyy orientation/documentation meeting:

**"cppyy is made up of the following packages: cppyy, cppyy-backend, and CPyCppyy."**

I have rephrases the section to clarify this, **please see if this is OK:**

cppyy is made up of the following packages: 
- A frontend: cppyy, 
- A backend: cppyy-backend, and 
- An extension: CPyCppyy.